### PR TITLE
Integration to threat_intel api.

### PR DESCRIPTION
**Pull request by bt3gl**
Moved the api.py files to api_deprecated.py from the following filters (that are imported from threat_intel package):

- shadowserver: api.py is exactly the same file as shadowserver.py so no further modifications were needed.

- virustotal: VirusTotalApi in threat_intel adds the resource_per_req parameter in the constructor, so a default value of 25 was added whenever this class is called in virustotal methods (which is in lookup_domain.py, _urls.py and _hashes.py).

- opendns: api.py is slightly different from opendns.py however only in private methods, and InvestigateApi class is the same.

Add threat_intel reqs to setup.py and requirements.txt. Add info to README.
Tests passed. Running osxcollector.py works.

Actions after the review: uncomment the api imports in the three filters, and delete the three api_deprecated.py files (I can do this).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,7 +15,13 @@ Armed with the forensic collection, an analyst can answer the question like:
 Yelp automates the analysis of most OSXCollector runs converting OSXCollector output into an easily readable and actionable summary of _just the suspicious stuff_.
 
 ## Performing Collection
-`osxcollector.py` is a single Python file that runs without any dependencies on a standard OSX machine. This makes it really easy to run collection on any machine - no fussing with brew, pip, config files, or environment variables. Just copy the single file onto the machine and run it.
+`osxcollector.py` is a single Python file that runs with very few dependencies on a standard OSX machine. This makes it really easy to run collection on any machine. Just run:
+
+```shell
```
Comment (ivanlei) [inline README.md]: I don't think this is necessary for collection.

File: osxcollector/output_filters/opendns/lookup_domains.py [hunk 2]
```diff
@@ -4,9 +4,11 @@
 #
 from collections import namedtuple
 
+from threat_intel.opendns import InvestigateApi
+
 from osxcollector.output_filters.base_filters.output_filter import run_filter_main
 from osxcollector.output_filters.base_filters.threat_feed import ThreatFeedFilter
-from osxcollector.output_filters.opendns.api import InvestigateApi
+#from osxcollector.output_filters.opendns.api import InvestigateApi
```
Comment (ivanlei) [inline osxcollector/output_filters/opendns/lookup_domains.py]: Please actually remove this line vs. commenting it out.

File: setup.py [hunk 3]
```diff
@@ -10,4 +10,7 @@
     author_email="opensource@yelp.com",
     description="A tool for answering \"How'd that malware get there?\"",
     packages=["osxcollector"],
+    install_requires=[
```
Comment (ivanlei) [inline setup.py]: For install_requires we're only including what's required for collection.
I think we only need `threat_intel` for analysis.

File: requirements.txt [hunk 4]
```diff
@@ -0,0 +1 @@
+threat_intel
```
Comment (ivanlei) [inline requirements.txt]: I think this can go in `requirements-dev.txt`

**Comment by ivanlei** [comment]
This is very cool.
Please do the deletes of unused code as part of this change.
I _think_ that this codebase may have some fixes around unicode that yelp/threat_intel is missing.
It's probably smart to diff the files as they're removed and make any updates to yelp/threat_intel.

File: README.md (hunk 1 above)
Comment (bt3gl) [inline README.md]: You are correct.

File: requirements.txt (hunk 4 above)
Comment (bt3gl) [inline requirements.txt]: I was under the impression that we needed threat_intel to run the collector, that's why I created a different requirement file (for running, not dev). But you are absolutely correct, so I'm moving it to -dev.

**Comment by ivanlei** [comment]
I think the following files can also be deleted:
https://github.com/Yelp/osxcollector/blob/master/osxcollector/output_filters/util/api_cache.py
https://github.com/Yelp/osxcollector/blob/master/osxcollector/output_filters/util/http.py

As well as `InvalidRequestError`:
https://github.com/Yelp/osxcollector/blob/master/osxcollector/output_filters/exceptions.py#L23

Probably `InvalidRequestError` belongs in `__all__`?

**Comment by bt3gl** [comment]
Apparently InvalidRequestError is only called by http.py (if you grep the entire folder, thats the only file that calls the class). I added exceptions to __all__, but I am not sure if need to import InvalidRequestError in __all__?

**Pull request merged**